### PR TITLE
Support SSR

### LINKED ISSUE #618: SSR in 4.4.0 crashes
> What is the current behavior?

Running this module on node.js (for example during ssr) results in: `ReferenceError: Element is not defined`

> What is the expected behavior?

Importing `react-transition-group` does not result in an error

The problem is this line:
https://github.com/reactjs/react-transition-group/blob/c8631587f9e1d18f5161f5490e6f5c225f06c6ae/src/Transition.js#L397

A [possible solution](https://stackoverflow.com/questions/48007326/what-is-the-correct-proptype-for-a-ref-in-react) is: `Element = typeof Element === 'undefined' ? function(){} : Element`

### PR DESCRIPTION
**Pull request by perrin4869**
Fixes #618 and adds regression tests to make sure the lib loads in node environment

### CODE REVIEW

File: src/Transition.js [hunk 1]
```diff
@@ -394,7 +394,7 @@ Transition.propTypes = {
    *     (see
    *     [test/CSSTransition-test.js](https://github.com/reactjs/react-transition-group/blob/13435f897b3ab71f6e19d724f145596f5910581c/test/CSSTransition-test.js#L362-L437)).
    */
-  nodeRef: PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
+  nodeRef: PropTypes.shape({ current: PropTypes.instanceOf(typeof Element === 'undefined' ? function() {} : Element) }),
```
Comment (oliviertassinari) [inline src/Transition.js:397]: Better not to rely on Element at all to support different windows http://tobyho.com/2011/01/28/checking-types-in-javascript/.

Comment (jquense) [reply src/Transition.js:397]:
```suggestion
  nodeRef: PropTypes.shape({ 
    current: typeof Element === 'undefined' 
      ? PropTypes.any
      : PropTypes.instanceOf(Element) 
   }),
```

File: src/Transition.js [hunk 2]
```diff
@@ -394,7 +394,11 @@ Transition.propTypes = {
    *     (see
    *     [test/CSSTransition-test.js](https://github.com/reactjs/react-transition-group/blob/13435f897b3ab71f6e19d724f145596f5910581c/test/CSSTransition-test.js#L362-L437)).
    */
-  nodeRef: PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
+  nodeRef: PropTypes.shape({ 
+    current: typeof Element === 'undefined' 
+      ? PropTypes.any
+      : PropTypes.instanceOf(Element)
```
Comment (oliviertassinari) [inline src/Transition.js:400]: Could we not rely on "Element" for this prop-type? My concern is equivalent to https://github.com/merceyz/typescript-to-proptypes/issues/3.

Comment (oliviertassinari) [reply src/Transition.js]: The problem can be noticed on https://github.com/mui-org/material-ui/issues/21858#issuecomment-661173090.

**Pull request merged**